gremlin-go/Dockerfile

```diff
@@ -15,23 +15,34 @@
 # specific language governing permissions and limitations
 # under the License.
 
-ARG GREMLIN_SERVER_VERSION=${version:-3.5.4}
-FROM tinkerpop/gremlin-server:${GREMLIN_SERVER_VERSION}-SNAPSHOT
+ARG GREMLIN_SERVER_VERSION
+FROM tinkerpop/gremlin-server:${GREMLIN_SERVER_VERSION:-3.5.4-SNAPSHOT}
 
 USER root
 RUN mkdir -p /opt
 WORKDIR /opt
 COPY gremlin-server/src/test /opt/test/
+COPY gremlin-go/docker/generate-all.groovy /opt/test/scripts/
 COPY gremlin-go/docker/docker-entrypoint.sh gremlin-go/docker/*.yaml /opt/
 RUN chmod 755 /opt/docker-entrypoint.sh
 
-# Setting to 3.5.3 does gives error even when using outside of Docker.
-ENV NEO4J_VERSION=${version:-3.5.2}
+# Installing dos2unix to avoid errors in running the entrypoint script on Windows machines where their
+# carriage return is \r\n instead of the \n needed for linux/unix containers
+RUN apk update && apk add dos2unix
+RUN dos2unix /opt/docker-entrypoint.sh && apk del dos2unix
+
+ARG NEO4J_VERSION
 # Installs Neo4j libraries to this image so that we can test variants with transactions,
 # but only only port 45940 is configured with the neo4j graph as the neo4j-empty.properties
 # is statically pointing at a temp directory and that space can only be accessed by one
 # graph at a time.
-RUN /opt/gremlin-server/bin/gremlin-server.sh install org.apache.tinkerpop neo4j-gremlin $NEO4J_VERSION
+RUN /opt/gremlin-server/bin/gremlin-server.sh install org.apache.tinkerpop neo4j-gremlin ${NEO4J_VERSION:-3.5.4}
```

Comment:
how does this work? 3.5.4 doesn't exist yet. it should be 3.5.4-SNAPSHOT (i.e. this should match whatever gremlin server is which should match whatever version gremlin-go is which should match whatever the root pom.xml is.)

Reply:
The -3.5.4 makes it default to 3.5.4, but can be changed via the "NEO4J_VERSION" build argument in docker-compose. I don't think there is a way to have the compose file dynamically parse the pom file, but I will make a run script to obtain the Gremlin server/Neo4j version and execute the docker-compose for now. Ultimately this will be resolved with the docker design works.